WPMedia/fusion-news-theme-blocks
# THEMES-1675: Overlay can close hamburger nav

## Description

This PR passes the close navigation function to the on click of the header nav chain overlay. This allows the nav to be closed by clicking on it. There were some issues with the aria labels I was looking into, but the fixes would be covered by the ticket THEMES-1666.

## Jira Ticket

- [THEMES-1675](https://arcpublishing.atlassian.net/browse/THEMES-1675)

## Acceptance Criteria

Nav can be closed by clicking the overlay.

## Test Steps

1. Checkout this branch `git checkout THEMES-1675`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/header-nav-chain-block`
3. Go to any page with the nav chain set on it.
4. Click the Sections button to open the hamburger nav.
5. Now click the semi-transparent overlay that appeared.
6. The nav should close and the overlay should disappear. This would be the same behavior as hitting the esc key or clicking the sections button again.

## Author Checklist

        please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist

_The reviewer of the PR should copy-paste this template into the review comments on review._



[THEMES-1675]: https://arcpublishing.atlassian.net/browse/THEMES-1675?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
